cpp/src/join/hash_join.cu

```diff
@@ -370,10 +370,20 @@ hash_join<Hasher>::hash_join(cudf::table_view const& build,
                              bool has_nulls,
                              cudf::null_equality compare_nulls,
                              rmm::cuda_stream_view stream)
+  : hash_join{build, has_nulls, compare_nulls, CUCO_DESIRED_LOAD_FACTOR, stream}
+{
+}
+
+template <typename Hasher>
+hash_join<Hasher>::hash_join(cudf::table_view const& build,
+                             bool has_nulls,
+                             cudf::null_equality compare_nulls,
+                             double load_factor,
+                             rmm::cuda_stream_view stream)
   : _has_nulls(has_nulls),
     _is_empty{build.num_rows() == 0},
     _nulls_equal{compare_nulls},
-    _hash_table{compute_hash_table_size(build.num_rows()),
+    _hash_table{compute_hash_table_size(build.num_rows(), load_factor * 100),
```

Comment:
We should move this after the `CUDF_EXPECTS` at L397-399

Reply:
https://godbolt.org/z/KbE9v7zo6

We need to initialize it using an initializer list since cuco hash tables are not default-constructible.

Reply:
Oh okay I see. Thanks for confirming. I guess, let's then add handling for load factor in cuco. Okay if the failing C++ tests are DISABLED for now and reenabled in a separate future PR.